analyzer/pom-analyzer/src/main/java/eu/fasten/analyzer/pomanalyzer/pom/DataExtractor.java

```diff
@@ -115,6 +115,45 @@ public String extractPackagingType(String groupId, String artifactId, String ver
         return packaging;
     }
 
+    /**
+     * Extracts project name from POM of certain Maven coordinate.
+     *
+     * @param groupId    groupId of the coordinate
+     * @param artifactId artifactId of the coordinate
+     * @param version    version of the coordinate
+     * @return Extracted project name as String
+     */
+    public String extractProjectName(String groupId, String artifactId, String version) {
+        String name = null;
+        try {
+            ByteArrayInputStream pomByteStream;
+            if ((groupId + ":" + artifactId + ":" + version).equals(this.mavenCoordinate)) {
+                pomByteStream = new ByteArrayInputStream(this.pomContents.getBytes());
+            } else {
+                pomByteStream = new ByteArrayInputStream(
+                        this.downloadPom(artifactId, groupId, version)
+                                .orElseThrow(FileNotFoundException::new).getBytes());
+            }
+            var pom = new SAXReader().read(pomByteStream).getRootElement();
+            if (this.resolutionMetadata == null || !this.resolutionMetadata.getLeft().equals(groupId + ":" + artifactId + ":" + version)) {
+                var metadata = this.extractDependencyResolutionMetadata(pom);
+                this.resolutionMetadata = new ImmutablePair<>(groupId + ":" + artifactId + ":" + version, metadata);
+            }
+            var properties = this.resolutionMetadata.getRight().getLeft();
```

Comment:
IDE complains about duplicate code here. This part is present in many methods of the class. Is there a way to extract it into a separate method and reuse it?

Reply:
Good point, I extracted those parts into separate methods